src/examples/java/com/amazonaws/crypto/examples/StringExample.java

```diff
@@ -68,8 +68,8 @@ public static void main(final String[] args) {
 
         // The SDK may add information to the encryption context, so we check to ensure
         // that all of our values are present
-        for (final Map.Entry<String, String> e : context.entrySet()) {
-            if (!e.getValue().equals(decryptResult.getEncryptionContext().get(e.getKey()))) {
+        for (final Map.Entry<String, String> e : decryptResult.getEncryptionContext().entrySet()) {
+            if (!e.getValue().equals(context.get(e.getKey())) && !e.getKey().equals(Constants.EC_PUBLIC_KEY_FIELD)) {
```

Comment:
could you please write the test case? Also whether this change is going to be backward compatible.